test/provisioner/provision_ibmcloud.go

```diff
@@ -339,7 +340,8 @@ func deleteSubnet() error {
 			return err
 		}
 	}
-
+	// Wait for sometime for subnet can be deleted
+	time.Sleep(30 * time.Second)
```

Comment:
Do you understand what is stopping the subnet from being deleted and is there some polling, or similar we can do instead of a sleep here? e.g. if the public gateway is stopping the subnet from being deleted do a wait for the public gateway to be deleted rather than this sleep?

Reply:
from the debug log when the error occur, it said:
```
00:51:25  time="2023-04-11T16:51:19Z" level=info msg="Cannot delete the subnet while it is in use by one or more instances. Please delete the network interfaces or their associated instances and retry: [instance 0797_6f64f449-3bd7-423a-b1bc-d548cdd20dd1 ]."
```
the instance is related the deleted IKS cluster, add some waiting time in DeleteCluster function.